neqo-http3/src/connection_client.rs

```diff
@@ -5458,4 +5458,23 @@ mod tests {
         }
         assert_eq!(count_responses, 2);
     }
+
+    #[test]
+    fn send_reserved_settings() {
+        for s in H3_RESERVED_SETTINGS {
+            let (mut client, mut server) = connect_only_transport();
+            // Create server control stream.
+            let control_stream = server.conn.stream_create(StreamType::UniDi).unwrap();
+            // Send the control stream type(0x0), a settings frame(0x4) of length 2.
+            let mut sf = vec![0x0, 0x4, 0x2];
```

Comment:
It might be better to use Encoder here and just write this out in full.  Even if that means a little more code.